core/cloudflow-streamlets/src/main/scala/cloudflow/streamlets/StreamletPort.scala

```diff
@@ -121,3 +136,21 @@ object RoundRobinPartitioner extends (Any => String) with Serializable {
    */
   def getInstance[T <: Any]: T => String = this
 }
+
+/**
+ * A default error handler. This error handler just logs bad message and skips it.
+ *
+ * Elements written to a [[CodecOutlet]] that uses this partitioner will be distributed in round-robin fashion across the topic partitions.
+ */
+object LoggingErrorHandler {
+
+  val logger = LoggerFactory.getLogger(this.getClass)
+
+  def processException[T](message: Array[Byte], result: Try[T]): Option[T] = result match {
+    case Success(value) => Some(value)
+    case Failure(e) =>
+      logger.error(s"Input data $message can not be transformed and will be skipped")
+      logger.error(s"Data transformation error id ${e.getMessage}")
```

Comment:
It's not transformation but decoding. just printing bytes is not a great idea. error id? Rather do `logger.error("Decode error, message will be skipped.", e)`

Reply:
Fixed